docs/dev/building-on-zksync/contracts/differences-with-ethereum.md

```diff
@@ -0,0 +1,192 @@
+
+On Ethereum, both `CREATE` and `CREATE2` accept the `initCode` of the contract with constructor parameters concatenated to it as its input. On zkSync, contract deployment is done via hashes, i.e. the operator receives the bytecodes themselves in the `factoryDeps` fields of the EIP712 transactions & the actual deployment is done by providing the contract’s hash to the `ContractDeployer` system contract.
+
+With this in mind, the compiler interprets the arguments of `create` as slightly unfilled arguments calldata to the `ContractDeployer` (the rest is filled with the compiler internally). For this to work the `datasize`/ `dataoffset` Yul instructions were adapted to return a constant size and the bytecode hash instead of bytecode. This means that in order for `create`/`create2` to work, the compiler needs to know the bytecode of the deployed contract beforehand.
+
+```solidity
+
+MyContract a = new MyContract();
+MyContract a = new MyContract{salt: ...}();
+
+```
+
+The following should also work as expected, but must be explicitly tested by the team in order to make sure that this is the case:
+
+```solidity
+bytes memory bytecode = type(MyContract).creationCode;
+assembly {
+    addr := create2(0, add(bytecode, 32), mload(bytecode), salt)
+} 
+```
+
+The following code will not work, because the compiler will not be aware of the bytecode beforehand:
+
+```solidity
+function myFactory(bytes memory bytecode) public {
+   assembly {
+      addr := create(0, add(bytecode, 0x20), mload(bytecode))
+   }
+}
+```
+
+Unfortunately it is not possible to distinguish between the cases above in compile time. And so it *highly recommended* to include tests for your factory regarding deploying the child contracts. 
+
+Also, note that for zkEVM bytecode we contain the different address derivation from Ethereum. The exact formulas are available in our SDK:
+
+```typescript
+export function create2Address(sender: Address, bytecodeHash: BytesLike, salt: BytesLike, input: BytesLike) {
+    const prefix = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('zksyncCreate2'));
+    const inputHash = ethers.utils.keccak256(input);
+    const addressBytes = ethers.utils
+        .keccak256(ethers.utils.concat([prefix, ethers.utils.zeroPad(sender, 32), salt, bytecodeHash, inputHash]))
+        .slice(26);
+    return ethers.utils.getAddress(addressBytes);
+}
+
+export function createAddress(sender: Address, senderNonce: BigNumberish) {
+    const prefix = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('zksyncCreate'));
+    const addressBytes = ethers.utils
+        .keccak256(
+            ethers.utils.concat([
+                prefix,
+                ethers.utils.zeroPad(sender, 32),
+                ethers.utils.zeroPad(ethers.utils.hexlify(senderNonce), 32)
+            ])
+        )
+        .slice(26);
+
+    return ethers.utils.getAddress(addressBytes);
+}
+```
+
+### Differences in `block.timestamp` and `block.number`
```

Comment:
We have 4-5 sections of differences.
Could we move them together and make of the same level?
Each opcode could have its own section, and the page could be started with a table presenting all affected opcodes with one-line short description and link to its section.